rfd/0073-public-image-registry.md

```diff
@@ -0,0 +1,249 @@
+---
+authors: Logan Davis (logan.davis@goteleport.com)
+state: draft
+---
+
+# RFD 73 - Public Image Registry
+
+## What
+Teleport images are currently hosted on [Quay.io](https://quay.io/organization/gravitational). This RFD proposes migrating public images from Quay to [Amazon ECR](https://aws.amazon.com/ecr/). 
+
+## Why
+As of August 1st, 2021 Quay.io no longer supports any other authentication provider other than Red Hat Single-Sign On. Users in the Gravitational organization on Quay must be manually removed when they leave Teleport which presents a potential security risk. Migrating to Amazon ECR will consolidate our infrastructure while also improving security with support for IAM policies and our existing SSO infrastructure.
+
+## Details
+Teleport will use Amazon ECR and Amazon ECR Public in order to host public container images. The multiple stage registry pipeline will allow Teleport to test and verify images internally before promoting them to our customers. As of authoring this RFD, Amazon ECR Public lacks support for [vulnerability scanning and tag immutabilty](https://github.com/aws/containers-roadmap/issues/1288). The two-stage pipeline will allow us to leverage these features in the internal repository before pushing to the public. 
+
+**What about name squatting on other container registry platforms?**
+
+A not yet finished RFD on third-party artifact mirroring will address concerns regarding name squatting on other container registry platforms. See [Artifact Mirroring](https://github.com/gravitational/teleport/commit/2262efbb25463ccc135553d43293f6d8aee22ba2).
+
+### Scope
+
+This RFD will focus on the infrastructure, security, and observability of the replacement registry. It will also include a deprecation and migration plan for the existing Quay.io repositories.
+
+#### In Scope
+* Infrastructure plans with [example terraform](#appendix-a-example-terraform)
+* Migration and Deprecation plan
+
+#### Out of Scope
+* Image Signing w/ Cosign
+* Mirroring of images to third-party registries. See [Artifact Mirroring](https://github.com/gravitational/teleport/commit/2262efbb25463ccc135553d43293f6d8aee22ba2).
```

Comment:
Mirroring to quay.io cannot be out of scope, since we must support extant deployments in the field.  I agree that mirroring to docker hub or other repos can be left out of scope for now.

Reply:
Good point, added an exception here: https://github.com/gravitational/teleport/pull/13165/commits/e5cd72e93f513f3a71c4f32bb9f75f7844efc650